zwave_js_server/util/lock.py

```diff
@@ -52,15 +53,34 @@ def _get_code_slots(
         except NotFoundError:
             return slots
 
+        code_slot = int(value.property_key)  # type: ignore
+
         # we know that code slots will always have a property key
         # that is an int, so we can ignore mypy
         slot = {
-            ATTR_CODE_SLOT: int(value.property_key),  # type: ignore
+            ATTR_CODE_SLOT: code_slot,
             ATTR_NAME: value.metadata.label,
             ATTR_IN_USE: status_value.value == CodeSlotStatus.ENABLED,
         }
         if include_usercode:
-            slot[ATTR_USERCODE] = value.value if value.value else None
+            if not value.value:
+                slot[ATTR_USERCODE] = None
+            elif value.value[0] != "{":
+                slot[ATTR_USERCODE] = value.value
+            else:
+                try:
+                    parsed_val = json.loads(value.value)
+                    if (
+                        "type" not in parsed_val
+                        or parsed_val["type"] != "Buffer"
+                        or "data" not in parsed_val
+                    ):
+                        raise ValueError("JSON string does not match expected schema")
+                    slot[ATTR_USERCODE] = "".join([chr(x) for x in parsed_val["data"]])
+                except (ValueError, json.decoder.JSONDecodeError) as err:
```

Comment:
`JSONDecodeError` is a subclass of `ValueError`.
```suggestion
                except ValueError as err:
```